### PR TITLE
step 1

### PR DESCRIPTION
**Pull request by excellencetechnologies**
Create a page to list do items, use a static array and display todo list.
Create component, service, and model

//explanation of how this class works is here
//Creating a class for the data
//https://angular.io/docs/ts/latest/guide/displaying-data.html

export class Todo {
	constructor(public id: number, public text: string,public active: boolean = false) {}
}

### CODE REVIEW

File: app/item/fetch-item.service.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+import {Item} from './item';
+import { ITEMS } from './item-list';
+import * as  _ from 'lodash';
+
+@Injectable()
+export class ItemService {
+    getItems() {
+        return Promise.resolve(ITEMS);
+    }
+    setItems(newItem: Item) {
```
Comment (upexcel) [inline app/item/fetch-item.service.ts]: what does this function do?

File: app/item/fetch-item.service.ts [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        var index = _.findIndex(ITEMS, function(o) {
+            return o.id == newItem.id;
+        });
+        console.log(index);
+        ITEMS[index] = newItem;
+    }
+    setComplete(newItem: Item) {
+        var index = _.findIndex(ITEMS, function(o) {
```
Comment (upexcel) [inline app/item/fetch-item.service.ts]: use map() function 

lodash.com/docs has many function, see map, filter etc dont use findindex for everything

File: app/item/item.template.html [hunk 3]
```diff
@@ -0,0 +1,28 @@
+<!DOCTYPE html>
```
Comment (upexcel) [inline app/item/item.template.html]: we do we have doctype, html, tead etc here? all this should be in index.html

why is this in template files??

File: app/todo.component.ts [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+    providers: [ItemService]
+})
+export class TodoComponent implements OnInit {
+    ngOnInit() {
+        this.getItems();
+    }
+    title = "My Todo App";
+    items: Item[];
+    selectedItem: Item;
+    constructor(private itemService: ItemService) {
```
Comment (upexcel) [inline app/todo.component.ts]: private variable  should have underscore. this is important

File: nbproject/private/private.properties [hunk 5]
```diff
@@ -1,4 +1,5 @@
 auxiliary.org-netbeans-modules-javascript-karma.config=karma.conf.js
```
Comment (upexcel) [inline nbproject/private/private.properties]: add nbproject to gitignore it should not come here

File: npm-debug.log [hunk 6]
```diff
@@ -1,39 +0,0 @@
-0 info it worked if it ends with ok
-1 verbose cli [ '/usr/bin/node', '/usr/bin/npm', 'start' ]
```
Comment (upexcel) [inline npm-debug.log]: this also in gitignore

File: package.json [hunk 7]
```diff
@@ -20,31 +20,30 @@
...
+    "@angular/compiler": "2.0.0-rc.1",
+    "@angular/core": "2.0.0-rc.1",
+    "@angular/http": "2.0.0-rc.1",
+    "@angular/platform-browser": "2.0.0-rc.1",
+    "@angular/platform-browser-dynamic": "2.0.0-rc.1",
+    "@angular/router": "2.0.0-rc.1",
+    "@angular/router-deprecated": "2.0.0-rc.1",
+    "@angular/upgrade": "2.0.0-rc.1",
+    "angular2-in-memory-web-api": "0.0.7",
+    "angular2-localstorage": "^0.2.0",
```
Comment (upexcel) [inline package.json]: remove localstorage

File: app/item/fetch-item.service.ts (hunk 1 above)
Comment (excellencetechnologies) [inline app/item/fetch-item.service.ts]: getItems return items array in ITEMS(static array list)

File: app/item/fetch-item.service.ts (hunk 2 above)
Comment (excellencetechnologies) [inline app/item/fetch-item.service.ts]: ok sir

File: app/item/item.template.html (hunk 3 above)
Comment (excellencetechnologies) [inline app/item/item.template.html]: will remove it.

File: nbproject/private/private.properties (hunk 5 above)
Comment (excellencetechnologies) [inline nbproject/private/private.properties]: removed.

File: npm-debug.log (hunk 6 above)
Comment (excellencetechnologies) [inline npm-debug.log]: ok sir

File: package.json (hunk 7 above)
Comment (excellencetechnologies) [inline package.json]: ok sir

File: app/add-item/add-item.component.ts [hunk 8]
```diff
@@ -0,0 +1,29 @@
...
+    constructor(private addService: AddItemService) {
+    }
+    onBack() {
+        window.history.back();
+    }
+    onSave(_task: string) {
+        this.addService.setItem(this._task).then(
+        item => this._item = item
+        );
+        alert("saved");
```
Comment (upexcel) [inline app/add-item/add-item.component.ts]: alert??

dont use alerts . and specially remove it from pull request

File: app/add-item/add-item.service.ts [hunk 9]
```diff
@@ -0,0 +1,20 @@
...
+import { ITEMS } from '../item/item-list';
+import * as  _ from 'lodash';
+
+@Injectable()
+export class AddItemService {
+    setItem(task: string) {
+        let obj = {
+            text: task,
+            active: false,
+            id: ITEMS.length + 1
```
Comment (upexcel) [inline app/add-item/add-item.service.ts]: does this work properly? is id unique always?

File: app/add-item/add-item.component.ts [hunk 10]
```diff
@@ -0,0 +1,29 @@
...
+    styleUrls: ['app/add-item/add-item.style.css'],
+    providers: [AddItemService]
+})
+export class AddItemComponent implements OnInit {
+    ngOnInit() {
+    }
+    public title = "Add Todo";
+    @Input() _task: string;
+     _item:Item;
+    constructor(private addService: AddItemService) {
```
Comment (upexcel) [inline app/add-item/add-item.component.ts]: private variables should have undescore, espcially services

File: app/item/fetch-item.service.ts [hunk 11]
```diff
@@ -0,0 +1,24 @@
...
+import { ITEMS } from './item-list';
+import * as  _ from 'lodash';
+
+@Injectable()
+export class ItemService {
+    getItems() {
+        return Promise.resolve(ITEMS);
+    }
+    setItems(newItem: Item) {
+        var index = _.findIndex(ITEMS, function(o) {
```
Comment (upexcel) [inline app/item/fetch-item.service.ts]: dont use findIndex

File: app/add-item/add-item.service.ts (hunk 9 above)
Comment (excellencetechnologies) [inline app/add-item/add-item.service.ts]: at present delete is not there.. so it will unique.

File: app/add-item/add-item.component.ts (hunk 10 above)
Comment (excellencetechnologies) [inline app/add-item/add-item.component.ts]: done

**Pull request merged**